### PR TITLE
HPCC-14700 Fix coverity and windows warnings

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
Windows compile complained about use of class X followed by struct X.
Coverity picked up that the (unused) RtlRecordSize class was missing a const on one function.

The second issue could have been caught by using override - except this file is also compiled with non c++11 (generated code), so that wouldn't work.

**Comment by ghalliday** [comment]
@shamser please review.

**Review by shamser — COMMENTED** [review]
@ghalliday This PR doesn't seem to be related to the jira ticket HPCC-14700 (Large numbers of variable length fields cause Long ECL Compile Times).  Please can you recheck that Jira ticket number or update the original ticket if necessary.

File: rtl/eclrtl/rtlrecord.hpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ class ECLRTL_API RtlRecordSize : public IRecordSize, public RtlCInterface
         return offsetCalculator.getRecordSize();
     }
 
-    virtual size32_t getFixedSize()
+    virtual size32_t getFixedSize() const
```
Comment (richardkchapman) [inline rtl/eclrtl/rtlrecord.hpp]: Is this issue new in 7.0.0? If not this should target 6.4.0

Comment (ghalliday) [inline rtl/eclrtl/rtlrecord.hpp]: Yes, that code is only is 7.0 - and is currently unused (which is why the problem wasn't discovered)

**Comment by ghalliday** [comment]
@shamser the JIRA is HPCC-14700 because both issues were introduced in the PR for that JIRA.

**Review by shamser — APPROVED** [review]
@richardkchapman Code reviewed ok.

**Pull request merged**